src/replication.c

```diff
@@ -1736,12 +1736,24 @@ void replicationSendNewlineToMaster(void) {
 }
 
 /* Callback used by emptyData() while flushing away old data to load
- * the new dataset received by the master and by discardTempDb()
- * after loading succeeded or failed. */
+ * the new dataset received by the master or to clear partial db if loading
+ * fails. */
 void replicationEmptyDbCallback(dict *d) {
     UNUSED(d);
     if (server.repl_state == REPL_STATE_TRANSFER)
         replicationSendNewlineToMaster();
+
+    processEventsWhileBlocked();
```

Comment:
maybe this call requires some throttling, not to be called too often?

Reply:
https://github.com/redis/redis/blob/3b1b1d14869c88d45f43fd4c333b9d1f54295fa9/src/dict.c#L701

It will be called from here. Once per 65535 keys, it seems good enough, probably no need to do extra throttling.